src/client/CodeEditor/getOrCreateEditor.ts

```diff
@@ -195,6 +198,69 @@ export const getOrCreateEditor = ({
 
   extensions.push(rotationIndicator);
 
+/*
+  extensions.push(autocompletion({
+    override: [tsComplete]
+  }))*/
+
+  // displays autocompletes
+  async function tsComplete(ctx) {
+
+    const compilerOptions = {};
+    const fsMap = await tsvfs.createDefaultMapFromCDN(
+      compilerOptions,
+      ts.version,
+      true,
+      ts
+    );
+    fsMap.set(name, text);
+    const system = tsvfs.createSystem(fsMap);
+    const env = tsvfs.createVirtualTypeScriptEnvironment(
+      system,
+      [name],
+      ts,
+      {
+      }
+    );
+
+    let tsCompletions = env.languageService.getCompletionsAtPosition(
```

Comment:
This call should happen inside the Web Worker, then the completions should be serialized and sent as a message back to the CodeMirror extension.